code/modules/mob/living/living.dm

```diff
@@ -423,6 +423,14 @@
 /mob/living/proc/get_organic_health()
 	return health
 
+/// Returns the amount of confusion on the mob.
+/// Use this over reading the confused variable directly
+/// if you are tying confusion to an effect.
+/mob/living/proc/get_confusion()
+	. = confused
+	for (var/datum/status_effect/confusion/confusion in has_status_effect_list(STATUS_EFFECT_CONFUSION))
+		. += confusion.strength
```

Comment:
Same here, this should be replaced by `if(HAS_TRAIT(src, TRAIT_CONFUSED))` in case someone needs to check if the mob is confused, and accessing directly the `confused` variable to check for current confusion strength, which the status effects should add to or subtract from.